src/redisearch_rs/inverted_index/src/lib.rs

```diff
@@ -136,6 +138,82 @@ pub struct RSIndexResult {
     pub weight: f64,
 }
 
+impl RSIndexResult {
+    pub fn numeric(num: f64) -> Self {
+        Self {
+            doc_id: 0,
+            dmd: std::ptr::null(),
+            field_mask: 0,
+            freq: 0,
+            offsets_sz: 0,
+            data: RSIndexResultData {
+                num: ManuallyDrop::new(RSNumericRecord(num)),
```

Comment:
What's the rationale for using `ManuallyDrop`  here?

Reply:
It's for the `union RSIndexResultData`.  It's basically to prevent a Rust `Drop` call on any union variant when the type goes out of scope.

https://github.com/RediSearch/RediSearch/blob/341f4d4e532782d6559ef5f8256dfb59bf6d0d17/src/redisearch_rs/inverted_index/src/lib.rs#L97-L101

![image](https://github.com/user-attachments/assets/69c6cf8b-ef95-4f98-b664-fa8e585564d2)